lib/queue.rb

```diff
@@ -1,31 +1,66 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = []
+    @max_size = 20
+    @front = -1
+    @rear = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if (@front == 0 && @rear == (@max_size - 1)) || 
+        (@rear == ((@front - 1) % (@max_size - 1)))
+          raise "No more space"
+    elsif @front == -1 # Queue is empty
+      # initialize front and rear
+      @front += 1
+      @rear += 1
+    elsif (@rear == (@max_size - 1) && @front != 0) # rear needs to wrap around
+      @rear = 0
+    else
+      @rear += 1
+    end
```

Comment:
This can be dried up with something like

```suggestion
    else
      @rear = (@rear + 1) % @max_size
    end
```